madara-prover-rpc-server/tests/test_run_bootloader.rs

```diff
@@ -0,0 +1,227 @@
+#[cfg(test)]
+mod tests {
+    use cairo_vm::air_private_input::{AirPrivateInput, AirPrivateInputSerializable};
+    use cairo_vm::cairo_run::CairoRunConfig;
+    use cairo_vm::hint_processor::builtin_hint_processor::bootloader::types::{
+        BootloaderConfig, BootloaderInput, PackedOutput, SimpleBootloaderInput, Task, TaskSpec,
+    };
+    use cairo_vm::hint_processor::builtin_hint_processor::builtin_hint_processor_definition::BuiltinHintProcessor;
+    use cairo_vm::hint_processor::hint_processor_definition::HintProcessor;
+    use cairo_vm::types::program::Program;
+    use cairo_vm::vm::errors::cairo_run_errors::CairoRunError;
+    use cairo_vm::vm::errors::vm_exception::VmException;
+    use cairo_vm::vm::runners::cairo_pie::CairoPie;
+    use cairo_vm::vm::runners::cairo_runner::CairoRunner;
+    use cairo_vm::vm::security::verify_secure_runner;
+    use cairo_vm::vm::vm_core::VirtualMachine;
+    use cairo_vm::{any_box, Felt252};
+    use madara_prover_common::models::PublicInput;
+    use madara_prover_common::toolkit::read_json_from_file;
+    use madara_prover_rpc_server::cairo::{extract_execution_artifacts, ExecutionArtifacts};
+    use rstest::{fixture, rstest};
+    use std::any::Any;
+    use std::collections::HashMap;
+    use std::path::Path;
+    use test_cases::{get_test_case_file_path, load_test_case_file};
+    use test_fixtures::{assert_memory_eq, assert_private_input_eq};
+
+    // Copied from cairo_run.rs and adapted to support injecting the bootloader input.
+    // TODO: check if modifying CairoRunConfig to specify custom variables is accepted upstream.
+    pub fn cairo_run(
```

Comment:
Is this the same as the other (non-test) one in `starknet_prover.rs`? (Could add a comment about why they're different if you actually need it)